src/ProjectAudioManager.cpp

```diff
@@ -520,9 +520,10 @@ void ProjectAudioManager::OnRecord(bool altAppearance)
             }
 
             existingTracks = ChooseExistingRecordingTracks(*p, false, options.rate);
-            t0 = std::max( t0, trackRange.max( &Track::GetEndTime ) );
+            if(!existingTracks.empty())
+                t0 = std::max( t0, trackRange.max( &Track::GetEndTime ) );
```

Comment:
I think this line changes behavior in the case that append-record must make a new wave tracks, so it's not really appending.

But my understanding of the description of issue #1265 is that this is not the case where change of behavior is intended.

Reply:
That's how i understood a task, maybe there is a problem with description. As i understand @jounih has confirmed that this is desired behaviour - recording should start from the point where play head is (does not depend on whether there an appropriate track for recording exist or not), the exception is an empty project - pressing the record button should create a new track with clip starting from beginning (because there could not be selection on a empty track, so behaviour stays consistent)

Reply:
I still have doubts about this line, being unnecessary for the stated description in the issue.

But this change will make a difference only in an unusual edge case, that wave tracks exit but none of them is chosen to record into.  Such a case might occur if all existing wave tracks are mono, but you must record stero, and so make a new track instead.

Reply:
The old behavior would be that recording would start at the latest of all end-times of existing tracks, or the left edge of the selection.  The new behavior would be to start recording always at the left edge of selection.

Reply:
But I think we can debate that edge case later.